src/core/common/log.hpp

```diff
@@ -49,7 +49,9 @@ namespace ot {
  *
  * This definition indicates whether or not logging is enabled.
  */
-#define OT_SHOULD_LOG (OPENTHREAD_CONFIG_LOG_OUTPUT != OPENTHREAD_CONFIG_LOG_OUTPUT_NONE)
+#define OT_SHOULD_LOG                                                     \
+    (OPENTHREAD_CONFIG_LOG_OUTPUT != OPENTHREAD_CONFIG_LOG_OUTPUT_NONE && \
+     OPENTHREAD_CONFIG_LOG_LEVEL > OT_LOG_LEVEL_NONE)
```

Comment:
As I said in the other comment, this is not corret and we should not add the `OPENTHREAD_CONFIG_LOG_LEVEL > OT_LOG_LEVEL_NONE` check:
- `CONFIG_LOG_LEVEL` specifies the intended level for OpenThread core modules only.
- `LogCert()` can still be used (regardless of `CONFIG_LOG_LEVEL`).
- Similarly, `otLogPlat()/otLogPlatArgs` (which are public OT APIs) can still be used by platform modules (and don't care about `LOG_LEVEL_CONFIG`).